From 86de2bd0527c8bf364cd9e10afa6ca1a1230b922 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 3 Sep 2014 15:15:51 -0400
Subject: [ethernet] bna: Replace large udelay() with mdelay()

Message-id: <1409757361-26781-4-git-send-email-ivecera@redhat.com>
Patchwork-id: 91165
O-Subject: [RHEL7.1 PATCH 03/13] bna: Replace large udelay() with mdelay()
Bugzilla: 1136912
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

BZ: #1136912

Upstream commit(s):
commit bc48bc806442114ea44f61d6b18e02c2bd6236fd
Author: Ben Hutchings <ben@decadent.org.uk>
Date:   Sun Mar 9 04:03:22 2014 +0000

    bna: Replace large udelay() with mdelay()

    udelay() does not work on some architectures for values above
    2000, in particular on ARM:

    ERROR: "__bad_udelay" [drivers/net/ethernet/brocade/bna/bna.ko] undefined!

    Reported-by: Vagrant Cascadian <vagrant@debian.org>
    Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/brocade/bna/bfa_ioc.c b/drivers/net/ethernet/brocade/bna/bfa_ioc.c
index 1803c39..354ae97 100644
--- a/drivers/net/ethernet/brocade/bna/bfa_ioc.c
+++ b/drivers/net/ethernet/brocade/bna/bfa_ioc.c
@@ -1704,7 +1704,7 @@ bfa_flash_sem_get(void __iomem *bar)
  while (!bfa_raw_sem_get(bar)) {
   if (--n <= 0)
    return BFA_STATUS_BADFLASH;
-  udelay(10000);
+  mdelay(10);
  }
  return BFA_STATUS_OK;
 }
-- 
1.7.1